django/db/migrations/autodetector.py

```diff
@@ -175,8 +176,10 @@ def _detect_changes(self, convert_apps=None, graph=None):
         # This avoids the same computation in generate_removed_indexes()
         # and generate_added_indexes().
         self.create_altered_indexes()
+        self.create_altered_constraints()
         # Generate index removal operations before field is removed
         self.generate_removed_indexes()
+        self.generate_removed_constraints()
```

Comment:
In the context of e.g. unique constraints depending on indexes to maybe remove constraints before removing indexes?